windows/src/desktop/kmshell/main/Keyman.System.UpdateCheckStorage.pas

```diff
@@ -49,4 +52,36 @@ class function TUpdateCheckStorage.LoadUpdateCacheData(var data: TUpdateCheckRes
     data.LoadFromFile(MetadataFilename, 'bundle', CKeymanVersionInfo.Version);
 end;
 
+class function TUpdateCheckStorage.HasKeyboardPackages(const data: TUpdateCheckResponse): Boolean;
+var
+  i : Integer;
+  fileName : string;
+  KeyboardRegex: TRegEx;
+begin
+  Result := False;
+  KeyboardRegex := TRegEx.Create('\.k..$');
+  for i := 0 to High(data.Packages) do
+  begin
+    fileName := data.Packages[i].FileName;
+    if KeyboardRegex.IsMatch(fileName) then
+      Result := True;
+  end;
+end;
+
+class function TUpdateCheckStorage.HasKeymanInstallFile(const data: TUpdateCheckResponse): Boolean;
+var
+  i : Integer;
+  fileName : string;
+  f: TSearchRec;
+begin
+  Result := False;
+  for i := 0 to High(data.Packages) do
+  begin
+    fileName := data.Packages[i].FileName;
+    if FindFirst(fileName + '*.exe', 0, f) = 0 then
+      Result := True;
+    System.SysUtils.FindClose(f);
```

Comment:
This seems strange; I didn't think the .exe was sent through as a package?

Reply:
The whole if was bogus. fixed here https://github.com/keymanapp/keyman/pull/12840/commits/dbaa7198a6cf981308a18fb88c9bad02f3ab3629